open_rarity/models/collection.py

```diff
@@ -47,13 +49,15 @@ class Collection:
         All trait names and values should be lowercased.
     name: A reference string only used for debugger log lines
 
+    We do not recommend resetting @tokens attribute after Collection initialization
+    as that will mess up cached property values:
+        has_numeric_attributes
+        get_token_standards
```

Comment:
Maybe a point of improvement for the future but, could we modify to use a factory pattern so that the `@tokens` attribute can't be reset? or just remove the setter for `tokens`?

So in the `__init__` we would have 
```python
self._tokens = tokens
```
and then the `tokens` prop would just be 

```python 
@property
def tokens(self) -> Token:
    return self._tokens
```

Reply:
Yeah I was thinking about this and never did it. Will add!